tests/conftest.py

```diff
@@ -208,50 +223,65 @@ def __init__(self, name=None, imgname=None, keeptb=False, fakeplatform=None):
                 if ctn.id == ctn_sw_id or ctn.name == ctn_sw_id:
                     ctn_sw_name = ctn.name
 
-            if sys.version_info < (3, 0):
-                (status, output) = commands.getstatusoutput("docker inspect --format '{{.State.Pid}}' %s" % ctn_sw_name)
-            else:
-                (status, output) = subprocess.getstatusoutput("docker inspect --format '{{.State.Pid}}' %s" % ctn_sw_name)
-            self.ctn_sw_pid = int(output)
-
-            # create virtual servers
-            self.servers = []
-            for i in range(32):
-                server = VirtualServer(ctn_sw_name, self.ctn_sw_pid, i)
-                self.servers.append(server)
+            if ctn_sw_name != "":
+                if sys.version_info < (3, 0):
+                    (status, output) = commands.getstatusoutput("docker inspect --format '{{.State.Pid}}' %s" % ctn_sw_name)
+                else:
+                    (status, output) = subprocess.getstatusoutput("docker inspect --format '{{.State.Pid}}' %s" % ctn_sw_name)
+                self.ctn_sw_pid = int(output)
 
-            self.mount = "/var/run/redis-vs/{}".format(ctn_sw_name)
+                # create virtual servers
+                self.servers = []
+                for i in range(32):
+                    server = VirtualServer(ctn_sw_name, self.ctn_sw_pid, i)
+                    self.servers.append(server)
 
+                self.mount = "/var/run/redis-vs/{}".format(ctn_sw_name)
+            else:
+                self.mount = "/var/run/redis-vs/{}".format(name)
             self.net_cleanup()
             self.ctn_restart()
         else:
             self.ctn_sw = self.client.containers.run('debian:jessie', privileged=True, detach=True,
-                    command="bash", stdin_open=True)
+                                                     command="bash", stdin_open=True)
             if sys.version_info < (3, 0):
                 (status, output) = commands.getstatusoutput("docker inspect --format '{{.State.Pid}}' %s" % self.ctn_sw.name)
             else:
                 (status, output) = subprocess.getstatusoutput("docker inspect --format '{{.State.Pid}}' %s" % self.ctn_sw.name)
             self.ctn_sw_pid = int(output)
-
             # create virtual server
             self.servers = []
             for i in range(32):
                 server = VirtualServer(self.ctn_sw.name, self.ctn_sw_pid, i)
                 self.servers.append(server)
 
+            if self.vct is not None:
+                self.vct_connect(newctnname)
+                time.sleep(2)
```

Comment:
any reason to sleep?

Reply:
On some of our test servers, the test_virtual_chassis ran much before the connectivity was up between supervisor and LC. 
I added this sleep much before I had a verify_chassis to pass after chassis creation. (line 1192).
With the verify_chassis being a better way to confirm virtual chassis is ready to run tests, this sleep is not required anymore.